plugins/pulp_docker/plugins/registry.py

```diff
@@ -303,6 +303,12 @@ def __init__(self, name, download_config, registry_url, working_dir):
 
         self.download_config = download_config
         self.registry_url = registry_url
+
+        # Use basic auth information only for retrieving tokens from auth server.
+        self.token_downloader = HTTPThreadedDownloader(self.download_config,
+                                                       AggregatingEventListener())
+        self.download_config.basic_auth_username = None
```

Comment:
I'm surprised these don't have an effect on the token downloader, since it's the same config object. I recommend making a copy with the ``copy`` module for safety.